pkg/ingester/wal.go

```diff
@@ -91,12 +96,22 @@ func newWAL(cfg WALConfig, userStatesFunc func() map[string]*userState, register
 	}
 
 	w := &walWrapper{
-		cfg:           cfg,
-		quit:          make(chan struct{}),
-		wal:           tsdbWAL,
-		getUserStates: userStatesFunc,
+		cfg:                   cfg,
+		quit:                  make(chan struct{}),
+		wal:                   tsdbWAL,
+		getUserStates:         userStatesFunc,
+		recordBuffer:          make([][]byte, 100),
+		recordBufferErrorChan: make([]chan error, 100),
```

Comment:
Why 100? It should be a constant too and reused.

Reply:
Still experimenting :) nothing is fixed here, not even the implementation. Latencies are still high.